### ISSUE TITLE
Loading inline scripts

### ISSUE THREAD

**Issue by RecuencoJones** [jenkinsci/JenkinsPipelineUnit#365]
### Dependencies

None

### Feature Request

Even though having all testing scripts as files under `src/main/jenkins` or at root (or wherever configured) is great, it might be sometimes convenient and more flexible to create the scripts inline in the test definition.

I've been testing this so far with one very basic example of a shared library, creating the following method in scope of a test class extending `BasePipelineTest`:

```groovy
Script loadInlineScript(String scriptText) {
    Objects.requireNonNull(binding, 'Binding cannot be null.')
    Objects.requireNonNull(helper.gse, 'GroovyScriptEngine is not initialized: Initialize the helper by calling init().')

    GroovyShell shell = new GroovyShell(helper.gse.getParentClassLoader(), binding, helper.gse.getConfig())
    Script script = shell.parse(scriptText)

    // make sure to set global vars after parsing the script as it will trigger library loads, otherwise library methods will be unregistered
    helper.setGlobalVars(binding)

    InterceptingGCL.interceptClassMethods(script.metaClass, helper, binding)

    return script
}
```

This could be easily added to `PipelineTestHelper` and proxied in `BasePipelineTest` just like `loadScript`.

This is how the test setup and method looks like:

```groovy
@Override
@Before
void setUp() throws Exception {
    super.setUp()

    def library = library()
                    .name('MySharedLibrary')
                    .retriever(projectSource())
                    .defaultVersion('<notNeeded>')
                    .targetPath('<notNeeded>')
                    .allowOverride(true)
                    .implicit(false)
                    .build()

    helper.registerSharedLibrary(library)
}

@Test
void 'should call echo with given subject from shared library wrapper' () {
    def script = loadInlineScript('''
        @Library('MySharedLibrary') _

        node {
            hello 'Jenkins'
        }
    ''')

    script.run()
    printCallStack()

    assertJobStatusSuccess()
}
```

If you think this could be interesting, I can prepare a PR for it 😄

**Comment by nre-ableton** [comment]
I personally don't have a use-case for this feature, but it is interesting. :smile: That said, I'm not opposed to a PR assuming that it is properly tested etc.

**Comment by red8888** [comment]
I was just going to ask how to do this. It would be great to have loadScript accept raw text or have a different loadInlineScript() method

**Comment by red8888** [comment]
@RecuencoJones I tried your loadInlineScript() function but it seems to hang forever on this part:
`Script script = shell.parse(scriptText)`

not sure why its hanging there

**Comment by RecuencoJones** [comment]
Hey! Sorry this thread got completely lost on my mail 😅

@nre-ableton I will try to follow up with a PR and some set of tests around it

@red8888 it's been a while without using it, let me test it tomorrow and I'll try to find if there's any other changes I did

**Comment by red8888** [comment]
It seems to hang if the scriptText includes a shared library reference `@Library('MySharedLibrary') _` 

that doesnt happen for me with the default loadScript method

**Comment by RecuencoJones** [comment]
Hi @red8888,

I'm also testing with a shared library and it doesn't seem to hang, even with such library references:

```
class MetaTest extends IntegrationTest {

    @Test
    void 'should use default properties for job' () {
        runInlineScript('''
        @Library('GlobalJenkinsLibrary') _

        meta.pipelines.withDefaults()
        ''')

        printCallStack()

        assertJobStatusSuccess()
    }
}
```

**Comment by red8888** [comment]
interesting not sure why its happening for me then

**Comment by nre-ableton** [comment]
Fixed by https://github.com/AbletonAppDev/eager/pull/162.

**Comment by nre-ableton** [comment]
Fixed by #382
